ls1intum/Artemis
# Feature/lazy loading

This PR activates lazy loading for a couple more modules to reduce the initial bundle size from around 2.5 MB to 1.5 MB. This forces improvements in the module structure and reduces the coupling between components.

The PR includes some minor additional changes.

### Checklist

**It is important to test whether the navigation to ALL pages works, so I would kindly ask everyone to help**

### Steps for Testing
1. Log in to Artemis
2. Navigate to any Artemis page and check if it works as expected

Fokus on:
- Code Editor and Markdown Editor
- Participation and submission for all exercise types
- Course Management
